chain/consensus/tspow/consensus.go

```diff
@@ -2,35 +2,232 @@ package tspow
 
 import (
 	"context"
+	"crypto/rand"
+	"fmt"
 
-	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/lib/async"
+	logging "github.com/ipfs/go-log/v2"
+	"golang.org/x/xerrors"
 
-	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
+	lapi "github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/build"
+	"github.com/filecoin-project/lotus/chain"
+	"github.com/filecoin-project/lotus/chain/beacon"
 	"github.com/filecoin-project/lotus/chain/consensus"
+	"github.com/filecoin-project/lotus/chain/stmgr"
 	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/lib/sigs"
+)
+
+const (
+	MaxDiffLookback = 70
+
+	// MaxHeightDrift is the epochs number to define blocks that will be rejected,
+	// if there are more than MaxHeightDrift epochs above the theoretical max height
+	// based on systime.
+	MaxHeightDrift = 5
```

Comment:
Instead of using constants, should we have a configuration where we define these?

Reply:
Where would you suggest to set this config parameter? (I think is a good idea to keep all these params in a config)

Reply:
I don't have a good suggestion at the moment, unfortunately. I think this needs to be part of a wider conversation regarding configuration parameters and command-line flags. Ideally, those would all be centralized on a single place. Right now, I have the feeling that they're scattered among a few different places.